sklearn/tests/test_cross_validation.py

```diff
@@ -483,6 +483,36 @@ def test_predefinedsplit_with_kfold_split():
     assert_array_equal(ps_test, kf_test)
 
 
+def test_shuffle_labels_out():
+    ys = [np.array([1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 3, 3]),
+          np.array([0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3]),
+          np.array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2]),
+          np.array([1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4]),
+          ]
+
+    for y in ys:
+        n_iter = 6
+        slo = cval.ShuffleLabelsOut(y, n_iter, test_size=0.33,
+                                    random_state=0)
+
+        # Make sure the repr works
+        repr(slo)
+
+        # Test that the length is correct
+        assert_equal(len(slo), n_iter)
+
+        for train, test in slo:
+            # First test: no train label is in the test set and vice versa
+            assert_false(np.any(np.in1d(y[train], np.unique(y[test]))))
+            assert_false(np.any(np.in1d(y[test], np.unique(y[train]))))
+
+            # Second test: train and test add up to all the data
+            assert_equal(y[train].size + y[test].size, y.size)
```

Comment:
I think you could reuse [`_check_is_partition`](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/cross_validation.py#L1107) and the check would be stronger